src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp

```diff
@@ -446,11 +446,14 @@ void SyncInferRequest::wait() {
             if (user_mem->get_allocation_type() == cldnn::allocation_type::cl_mem && output_memory->get_allocation_type() != cldnn::allocation_type::cl_mem) {
                 // WA: Copy between cl_mem and usm memory may fail for some reason (driver bug?)
                 // so this explicit memcpy is used to provide correct output for cl_mem output in dynamic cases
-                cldnn::mem_lock<uint8_t, cldnn::mem_lock_type::write> lock_dst(user_mem, stream);
-                cldnn::mem_lock<uint8_t, cldnn::mem_lock_type::read> lock_src(output_memory, stream);
+                auto blocking = (stream.get_queue_type() == QueueTypes::in_order) ? false : true;
+                cldnn::mem_lock<uint8_t, cldnn::mem_lock_type::read> lock_dst(user_mem, stream, blocking);
+                cldnn::mem_lock<uint8_t, cldnn::mem_lock_type::read> lock_src(output_memory, stream, blocking);
```

Comment:
But isn't  it better to use 
user_mem->copy_from(stream, output_memory, blocking) ?

Reply:
updated as suggested and removed unnecessary code changes